shared/metrics/src/main/java/io/pravega/shared/MetricsNames.java

```diff
@@ -269,33 +233,24 @@ public static String joinWithDot(String element1, String element2) {
     }
 
     /**
-     * Create an MetricKey based on metric name, metric type and tags associated.
-     *
-     * e.g.
-     * for flat metric name "append_count.6" (without tag), its cache key is:
-     *   "append_count.6.Counter", and its registry key is "append_count.6.Counter".
-     * for metric name "append_count" with tag "container=6", the cache key is:
-     *   "append_count.6.Counter", but the registry key is "append_count".
+     * Create an MetricKey object based on metric name, metric type and tags associated.
+     * The MetricKey object contains cache key for cache lookup and registry key for registry lookup.
      *
      * @param metric the metric name.
-     * @param typeSuffix the metric type suffix, e.g. ".Counter", ".Gauge", ".Meter"
      * @param tags the tag(s) associated with the metric.
-     * @return the MetricKey object which holds both cache lookup key and metric registry key.
+     * @return the MetricKey object contains cache lookup key and metric registry key.
      */
-    public static MetricKey metricKey(String metric, String typeSuffix, String... tags) {
+    public static MetricKey metricKey(String metric, String... tags) {
 
-        StringBuilder sb = new StringBuilder(metric);
-        if (tags == null || tags.length == 0) {  //no tags: append suffix to form cache & registry key
-            sb.append(typeSuffix);
-            String key = sb.toString();
-            return new MetricKey(key, key);
-        } else { //tags: append tag values and suffix to form cache key, original name is registry key
+        if (tags == null || tags.length == 0) {  //without tags: the original metric name is cache and registry key
```

Comment:
This comment seems like it's missing something

Reply:
Fixed.